presto-tests/src/main/java/com/facebook/presto/tests/FeatureSelection.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.google.common.collect.ImmutableSet;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static java.util.Objects.requireNonNull;
+
+public final class FeatureSelection
+{
+    public static FeatureSelection all()
+    {
+        return allFeatures().build();
+    }
+
+    public static Builder allFeatures()
+    {
+        return new Builder();
+    }
+
+    public static Builder features(TestedFeature... features)
+    {
+        requireNonNull(features);
+        return new Builder().including(features);
+    }
+
+    public static FeatureSelection excluding(TestedFeature... features)
+    {
+        requireNonNull(features);
+        return new Builder().excluding(features);
+    }
+
+    private final Optional<Set<TestedFeature>> includes;
+    private final Optional<Set<TestedFeature>> excludes;
+
+    private FeatureSelection(Optional<Set<TestedFeature>> includes, Optional<Set<TestedFeature>> excludes)
+    {
+        requireNonNull(includes, "includes is null");
+        requireNonNull(excludes, "excludes is null");
+
+        this.includes = includes.map(ImmutableSet::copyOf);
+        this.excludes = excludes.map(ImmutableSet::copyOf);
+    }
+
+    public boolean isSelected(TestedFeature feature)
+    {
+        requireNonNull(feature);
+        return areSelected(ImmutableSet.of(feature));
+    }
+
+    public boolean areSelected(Set<TestedFeature> features)
+    {
+        requireNonNull(features);
+
+        if (excludes.isPresent() && excludes.get().stream().anyMatch(features::contains)) {
+            return false;
```

Comment:
if `excludes.get().contains(features.iterator().next().getFeatures().iterator().next())`, you should also return `false`

(i.e. if `excludes` contains `CREATE_TABLE`, then `isSelected(METADATA)` should return `false`)